sympy/polys/numberfields/galoisgroups.py

```diff
@@ -0,0 +1,808 @@
+"""Galois groups of polynomials. """
+
+import random
+
+from sympy.core.evalf import (
+    evalf, fastlog, _evalf_with_bounded_error, quad_to_mpmath,
+)
+from sympy.core.numbers import Integer
+from sympy.core.symbol import Dummy, symbols
+from sympy.functions.elementary.miscellaneous import sqrt
+from sympy.polys.domains import ZZ
+from sympy.polys.densebasic import dup_random
+from sympy.polys.densetools import dup_eval
+from sympy.polys.euclidtools import dup_discriminant
+from sympy.polys.factortools import dup_factor_list
+from sympy.polys.numberfields.utilities import coeff_search
+from sympy.polys.polytools import Poly
+from sympy.polys.sqfreetools import dup_sqf_p
+from sympy.utilities.lambdify import lambdify
+
+from mpmath import MPContext
+
+
+class GaloisGroupException(Exception):
+    ...
+
+
+class ResolventException(GaloisGroupException):
+    ...
+
+
+class MaxTriesException(GaloisGroupException):
+    ...
+
+
+class Resolvent:
```

Comment:
Do we need to have a new class to solve this problem?
I'd like the usage of objects to be minimal as possible. If it can be replaced by functions and compositions of primitive types (like list, dict, tuple), it is often trivial to define the whole object for adhoc purposes, and make the library more inflexible to use, unless you have defined some calculus by polymorphism.